Sources/PostgreSQL/Connection/PostgreSQLConnection+TCP.swift

```diff
@@ -1,11 +1,13 @@
 import Async
 import NIO
+import NIOOpenSSL
 
 extension PostgreSQLConnection {
     /// Connects to a Redis server using a TCP socket.
     public static func connect(
         hostname: String = "localhost",
         port: Int = 5432,
+        tlsConfiguration: TLSConfiguration? = nil,
```

Comment:
The way I did this with HTTPClient was to wrap the different schemes in a pseudo-enum. I think this would be a solid solution here as well.

Something like:
```swift
public struct PostgreSQLScheme {
    internal enum Storage {
        case plainText
        case tls(TLSConfiguration?)
    }

    internal let storage: Storage

    // lots of conveniences here 
}
```

That way if we need to support more methods in the future, we don't bloat the connect method. We can also provide convenience extensions to that struct easier.